src/components/AppBar/AppBar.test.js

```diff
@@ -0,0 +1,118 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import { shallow } from "enzyme";
+import PropTypes from "prop-types";
+import { assertPropTypes } from "check-prop-types";
+
+import AppBar from "./AppBar";
+import themes from "../common/themes";
+import { shadow } from "../common";
+
+describe("Component: AppBar", () => {
+
+  const testProps = {
+    fixed: true,
+    className: "my-class",
+    style: { color: "red" },
+    children: "foo",
+    shadow: true,
+  }
+
+  it("warns if props are missing or the wrong type", () => {
+    expect(() => { assertPropTypes(AppBar.propTypes, { ...testProps, fixed: 123 }, 'prop', AppBar.fixed); }).toThrow();
+    expect(() => { assertPropTypes(AppBar.propTypes, { ...testProps, shadow: 123 }, 'prop', AppBar.fixed); }).toThrow();
+    expect(() => { assertPropTypes(AppBar.propTypes, { ...testProps, children: undefined }, 'prop', AppBar.children); }).toThrow();
+    expect(() => { assertPropTypes(AppBar.propTypes, { ...testProps, className: 123 }, 'prop', AppBar.className); }).toThrow();
+    expect(() => { assertPropTypes(AppBar.propTypes, { ...testProps, style: "red" }, 'prop', AppBar.style); }).toThrow();
+  })
+
+
+  it("renders", () => {
+    const wrapper = shallow(<AppBar />);
+    expect(wrapper.find("AppBar__StyledAppBar").length).toBe(1);
+  });
+
+  it("renders children", () => {
+    const wrapper = shallow(<AppBar>text</AppBar>);
+    expect(wrapper.text()).toBe("text");
+  });
+
+  it("renders className", () => {
+    const wrapper = shallow(<AppBar className="my-class">text</AppBar>);
+    expect(wrapper.props().className).toBe("my-class");
+  });
+
+  it("renders style object", () => {
+    const wrapper = shallow(
+      <AppBar
+        href="#"
+        className="my-class"
+        style={{ color: "blue" }}
+      >
+        text
+      </AppBar>
+    );
+    expect(wrapper.props().style.color).toBe("blue");
+  })
+
+  it("renders other props passed", () => {
+    const wrapper = shallow(
+      <AppBar
+        href="#"
+        className="my-class"
+        style={{ color: "blue" }}
+        a={1}
+        b={2}
+        c={3}
+      >
+        text
+      </AppBar>
+    );
+    expect(wrapper.props().a).toBe(1);
+    expect(wrapper.props().b).toBe(2);
+    expect(wrapper.props().c).toBe(3);
+  });
+
+  it("renders default props", () => {
+    const wrapper = shallow(<AppBar />);
+    expect(wrapper.props().shadow).toBe(true);
+    expect(wrapper.props().fixed).toBe(true);
+  });
+
+  it("renders default component styles", () => {
+    const tree = renderer.create(<AppBar theme={themes.default} />).toJSON();
```

Comment:
Not sure if these kind of tests are valuable

Reply:
Snapshot tests aren't valuable at all, better do some screenshot testing

Reply:
I was referring to the highlighted test though: "renders default component styles"
I do personally find snapshot tests valuable.

Reply:
The readme for jest-styled-components had it as an [example](https://github.com/styled-components/jest-styled-components#tohavestylerule) in their docs, so thought I'd follow suit. Not sure how valuable they are either tbh. Maybe just testing the interpolated styles is enough? I'd be happy to add some screenshot tests as well.

Reply:
Tbh I don't know if we should keep this kind of tests. (I will have to do some research 😂). Lets leave it like it is for now and we will get back to this later